libsolidity/interface/AssemblyStack.cpp

```diff
@@ -117,7 +117,7 @@ MachineAssemblyObject AssemblyStack::assemble(Machine _machine) const
 	{
 		MachineAssemblyObject object;
 		julia::EVMAssembly assembly(true);
-		julia::CodeTransform(assembly, *m_analysisInfo, m_language == Language::Yul, true)(*m_parserResult);
+		julia::CodeTransform(assembly, *m_analysisInfo, *m_parserResult, m_language, true, m_language == Language::Yul, true)(*m_parserResult);
```

Comment:
the first `true` should be replaced by: m_language is Yul or StrictAssembly.

Reply:
Perhaps it might be better to rather pass `m_language` directly into the CodeTransform, then it can figure out the two switches `m_yul` and `m_stackOpt` itself.

Reply:
the "true" is not the one I passed, I actually passed the m_language and set the  m_allowStackOpt in constructor. But I only enable it for StrictAssembly now. Shall I also enable it for YUL?

Reply:
Yes, yul and strict assembly.

Reply:
:+1: thanks